src/Logic.hs

```diff
@@ -480,6 +482,31 @@ handlePullRequestEdited prId newTitle newBaseBranch state =
     -- Do nothing if the pull request is not present.
     Nothing -> pure state
 
+-- | If there is a push to a branch
+--   then update the integration status for the pullrequests
+--   that are being integrated and target the pushed branch.
+handleTargetChanged
+  :: forall es. (Action :> es, RetrieveEnvironment :> es)
+  => BaseBranch
+  -> ProjectState
+  -> Eff es ProjectState
+handleTargetChanged (BaseBranch branch) state | Text.isPrefixOf "refs/heads/" branch = pure $ Pr.updatePullRequests update state
+  where
+    update pr
+      | prIsIntegrating pr
+      && Pr.baseBranch pr == BaseBranch (Text.drop 11 branch)
+      = pr
+      { Pr.integrationStatus = Outdated
+      , Pr.needsFeedback = True}
+    update pr = pr
+handleTargetChanged _ state = pure state
+
+prIsIntegrating :: PullRequest -> Bool
+prIsIntegrating pr = case Pr.classifyPullRequest pr of
+                      PrStatusBuildPending -> True
+                      PrStatusBuildStarted _ -> True
+                      _ -> False
```

Comment:
I this is more appropriate in `Project.hs`.

Reply:
I found out the function actually already existed.